lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -86,37 +92,55 @@ def configure(conf)
         end
       end
 
-      unless @use_metadata_service
-        unless @project_id && @zone && @vm_id
-          raise Fluent::ConfigError,
-              ('Please specify "project_id", "zone" and "vm_id" if you set "use_metadata_service" to false')
+      if @use_metadata_service
+        if @project_id.nil?
+          begin
+            @project_id = fetch_metadata('project/project-id')
+          rescue
+            @project_id = nil
+          end
+        end
+        if @zone.nil?
+          begin
+            fully_qualified_zone = fetch_metadata('instance/zone')
+            @zone = fully_qualified_zone.rpartition('/')[2]
+          rescue
+            @zone = nil
+          end
+        end
+        if @vm_id.nil?
+          begin
+            @vm_id = fetch_metadata('instance/id')
+          rescue
+            vm_id = nil
+          end
         end
       end
-    end
-
-    def start
-      super
-
-      init_api_client()
-
-      @successful_call = false
-      @timenanos_warning = false
 
-      if @use_metadata_service
-        # Grab metadata about the Google Compute Engine instance that we're on.
-        @project_id = fetch_metadata('project/project-id')
-        fully_qualified_zone = fetch_metadata('instance/zone')
-        @zone = fully_qualified_zone.rpartition('/')[2]
-        @vm_id = fetch_metadata('instance/id')
+      # all metadata parameters must now be set
+      unless @project_id && @zone && @vm_id
+        missing = []
+        missing << "project_id" unless @project_id
+        missing << "zone" unless @zone
+        missing << "vm_id" unless @vm_id
+        raise Fluent::ConfigError,
+          ('Unable to obtain metadata parameters: ' + missing.join(' '))
       end
+
       # TODO: Send instance tags and/or hostname with the logs as well?
       @common_labels = {}
 
       # If this is running on a Managed VM, grab the relevant App Engine
       # metadata as well.
       # TODO: Add config options for these to allow for running outside GCE?
-      attributes_string = @use_metadata_service ?
-          fetch_metadata('instance/attributes/') : ""
+      attributes_string = ""
+      if (@use_metadata_service)
+        begin
+          attributes_string = fetch_metadata('instance/attributes/')
+        rescue
```

Comment:
this is the same basic idea, if there is no metadata service then we just proceed as if no attributes are set.  But, I think I will re-structure things a bit so instead of having 'rescue' here I will remember whether we have a metadata service or not, and only try to fetch if so.